internal/dbmodel/job_tracker.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2025 Canonical.
+
+package dbmodel
+
+import (
+	"errors"
+	"time"
+
+	"github.com/google/uuid"
+	"gorm.io/gorm"
+)
+
+// JobStatus represents the status of a job in the job tracker.
+// The status can be one of the following:
+// [statusPending], [StatusRunning], [StatusSuccessful], or [statusFailed].
+// It is stored as a custom enum in the database.
+// StatusFailed must be set using the [JobTrackerEntry.SetFailed] method.
+type JobStatus string
+
+const (
+	StatusRunning    JobStatus = "running"
+	StatusSuccessful JobStatus = "successful"
+	StatusPending    JobStatus = "pending"
+	StatusFailed     JobStatus = "failed"
+)
+
+// JobTrackerEntry represents a job that is being tracked within an instance of JIMM.
+type JobTrackerEntry struct {
+	// JobID is the unique identifier for the job. This is not to be set manually, please
+	// use the constructor [NewJobTrackerEntry] instead.
+	JobID uuid.UUID `gorm:"type:uuid;primaryKey"`
+	// JobType is the type of the job, e.g., bootstrap, destroy, etc.
+	JobType string `gorm:"type:varchar(128);not null"`
+	// Stop signals whether the job should be stopped. This is set to true when the job is
+	// requested to be stopped, but it does not mean the job has stopped yet.
+	// This can be determined by the Status field.
+	Stop bool `gorm:"not null;default:false"`
+	// Status holds the current status of the job.
+	Status JobStatus `gorm:"type:job_tracker_status;not null;default:'pending'"`
+	// Error holds any error message associated with the job. Not to be set manually
+	// and will only ever be set if the status is [StatusFailed].
+	Error string `gorm:"type:text"`
+
+	CreatedAt time.Time
```

Comment:
just include `gorm.Model`, but consider if we want `DeletedAt`

Reply:
Can't. We're not using ID, we're using JobID and UUIDs.